bika/lims/browser/analysisrequest/workflow.py

```diff
@@ -470,10 +470,13 @@ def workflow_action_publish(self):
         # AR publish preview
         uids = self.request.form.get('uids')
         if uids:
-            items=",".join(uids)
-            self.request.response.redirect(self.context.portal_url() + "analysisrequests/publish?items="+items)
+            items = ",".join(uids)
+            self.request.response.redirect(self.context.portal_url() +
+                                           "/analysisrequests/publish?items="
+                                           + items)
         else:
-            self.request.response.redirect(self.context.absolute_url() + "/publish")
+            self.request.response.redirect(self.context.absolute_url() +
+                                           "/publish")
```

Comment:
Makes the code simpler and allows to unify criterias (just always use the same url for publishing):

    uids = self.request.form.get('uids', [self.context.UID])
    items = ",".join(uids)
    self.request.response.redirect(self.context.portal_url() + "/analysisrequests/publish?items=" + items)